File: frontend/frontend_candidat/src/app/_MonClasses/OfferEmploi.ts
import { Departement } from "./dep";

export class OfferEmploi{
    id: number; // Changed from private Long to public number
    titre: string;
    resumePoste: string; // Changed from private String to public string
    typeOfferEmploi: string;
    location: string;
    salMax: number; // Changed from private double to public number
    salMin: number; // Changed from private double to public number
    dateCreation: string; // Kept as Date type
    dateExpiraton: string; // Kept as Date type
    etat: boolean;
    enattent: boolean;
    departement:Departement
}

File: backend/src/main/java/com/pfe/model/OfferEmploi.java
package com.pfe.model;

import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class OfferEmploi {
@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private Long id;
private String titre;
@Column(length = 9999)
 private String resumePoste;
private String typeOfferEmploi;
private String location;
private double salMax , salMin;
private Date dateCreation;
private Date dateExpiraton;
private boolean etat;
 private boolean enattent;
 

@ManyToOne
@JoinColumn(name = "id_dep") 
private Departement departement;
}
